src/Bundle/Maker/Factory/DoctrineObjectDefaultPropertiesGuesser.php

```diff
@@ -56,66 +57,125 @@ public function __invoke(MakeFactoryData $makeFactoryData, bool $allFields): voi
+
+            if (!$type) {
+                continue;
+            }
 
+            if (!$allFields && $type->isNullable()) {
                 continue;
             }
 
-            $factory = new \ReflectionClass($factoryClass);
-            $makeFactoryData->addUse($factory->getName());
-            $makeFactoryData->addDefaultProperty(\lcfirst($fieldName), "{$factory->getShortName()}::new(),");
+            if (!$type->getClassName()) {
+                continue;
+            }
+
+            /** @phpstan-ignore-next-line */
+            $this->addDefaultValueUsingFactory($makeFactoryData, $fieldName, $type->getClassName());
         }
     }
 
-    private function guessDefaultValueForScalarFields(MakeFactoryData $makeFactoryData, ODMClassMetadata|ORMClassMetadata $metadata, string $dbType, bool $allFields): void
+    private function guessDefaultValueForScalarFields(MakeFactoryData $makeFactoryData, ODMClassMetadata|ORMClassMetadata $metadata, bool $allFields): void
     {
         $ids = $metadata->getIdentifierFieldNames();
 
         foreach ($metadata->fieldMappings as $property) {
+            if ($property['embedded'] ?? false) {
+                // ODM embedded fields are present in both fieldMappings and associationMappings
+                continue;
+            }
+
+            $fieldName = $property['fieldName'];
+
+            if (\str_contains($fieldName, '.')) {
+                // this is a "subfield" of an ORM embeddable field.
+                continue;
+            }
+
             // ignore identifiers and nullable fields
-            if ((!$allFields && ($property['nullable'] ?? false)) || \in_array($property['fieldName'], $ids, true)) {
+            if ((!$allFields && ($property['nullable'] ?? false)) || \in_array($fieldName, $ids, true)) {
                 continue;
             }
 
             $type = \mb_strtoupper($property['type']);
-            $value = "null, // TODO add {$type} {$dbType} type manually";
+            $value = "null, // TODO add {$type} type manually";
             $length = $property['length'] ?? '';
 
             if (\array_key_exists($type, self::DEFAULTS)) {
                 $value = self::DEFAULTS[$type];
             }
 
-            $makeFactoryData->addDefaultProperty($property['fieldName'], \str_replace('{length}', (string) $length, $value));
+            $makeFactoryData->addDefaultProperty($fieldName, \str_replace('{length}', (string) $length, $value));
         }
     }
+
+    /** @param class-string $fieldClass */
+    private function addDefaultValueUsingFactory(MakeFactoryData $makeFactoryData, string $fieldName, string $fieldClass, bool $isMultiple = false): void
+    {
+        if (!$factoryClass = $this->factoryFinder->getFactoryForClass($fieldClass)) {
+            $makeFactoryData->addDefaultProperty(\lcfirst($fieldName), "null, // TODO add {$fieldClass} type manually");
+
+            return;
+        }
+
+        $factoryMethod = $isMultiple ? 'new()->many(5)' : 'new()';
```

Comment:
`5` is a totally arbitrary int...

Reply:
I'm thinking we should not have `--all-fields` populate `OneToMany` relationships.

This can cause problems when testing _count_'s of entities in the db. You can get a greater count than you expect. @OskarStark ran into this problem just the other day.

WDYT?

Reply:
Yes it can create side effects, because it can also nest many more entities downwards

Reply:
what about if we say `Factory::new(['propertyName' => $this])`?

Reply:
However, this is what we do in this test:
https://github.com/zenstruck/foundry/blob/f43b06708608f9f46691183d4417eebe967fbc94/tests/Functional/ORMModelFactoryTest.php#L65

even if I add this code, the test still passes:
```php
        foreach (CommentFactory::all() as $comment) {
            self::assertSame($post->object(), $comment->getPost());
        }
```

in which ways this code could be problematic?

Reply:
It's totally legit to create them this way in a test. But when say 10 are added via the `getDefaults()`. In a test, you do something that creates 2 entities, then assert the count is 2, the actual count could be 12 (or more depending on your setup).

Sorry, I'm having a hard time explaining.

Reply:
We are all on the same page now. We're going to restrict `--all-fields` to _nullable-non-relationships_.